src/transform3d.rs

```diff
@@ -740,6 +742,36 @@ where
             self.transform_point2d(point2(min.x, max.y))?,
         ]))
     }
+
+    /// Returns a 2d box that encompasses the result of transforming the given box by this
+    /// transform, if the transform makes sense for it, or `None` otherwise.
+    pub fn outer_transformed_box2d(&self, b: &Box2D<T, Src>) -> Option<Box2D<T, Dst>>
+    where
+        T: Sub<Output = T> + Div<Output = T> + Zero + PartialOrd,
+    {
+        Some(Box2D::from_points(&[
+            self.transform_point2d(b.min)?,
+            self.transform_point2d(b.max)?,
+            self.transform_point2d(point2(b.max.x, b.min.y))?,
+            self.transform_point2d(point2(b.min.x, b.max.y))?,
+        ]))
+    }
+
+    /// Returns a 3d box that encompasses the result of transforming the given box by this
+    /// transform, if the transform makes sense for it, or `None` otherwise.
+    pub fn outer_transformed_box3d(&self, b: &Box3D<T, Src>) -> Option<Box3D<T, Dst>>
+    where
+        T: Sub<Output = T> + Div<Output = T> + Zero + PartialOrd,
+    {
+        Some(Box3D::from_points(&[
+            self.transform_point3d(b.min)?,
+            self.transform_point3d(b.max)?,
+            self.transform_point3d(point3(b.max.x, b.min.y, b.min.y))?,
+            self.transform_point3d(point3(b.min.x, b.max.y, b.min.y))?,
+            self.transform_point3d(point3(b.max.x, b.min.y, b.max.y))?,
+            self.transform_point3d(point3(b.min.x, b.max.y, b.max.y))?,
```

Comment:
why are we using only 6 points here instead of the full 8?

Reply:
eeeerrrr to prove to the world that code reviews are a good thing.

Reply:
Ouch, even worse than that, I wrote `y` instead of `z`.

Reply:
Fixed.